server/src/main/java/org/elasticsearch/search/runtime/StringScriptFieldTermQuery.java

```diff
@@ -8,30 +8,46 @@
 
 package org.elasticsearch.search.runtime;
 
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.LeafReaderContext;
 import org.apache.lucene.index.Term;
+import org.apache.lucene.search.Query;
 import org.apache.lucene.search.QueryVisitor;
 import org.elasticsearch.script.Script;
 import org.elasticsearch.script.StringFieldScript;
 
+import java.io.IOException;
 import java.util.List;
 import java.util.Objects;
+import java.util.function.Function;
 
 public class StringScriptFieldTermQuery extends AbstractStringScriptFieldQuery {
```

Comment:
Will there be prefix query support ay some point? (not suggesting adding to this PR - just curious).

Reply:
It's possible. We'd have to detect a `substring` with a leading constant of 0. But we can do that. It'd be fairly similar for a grok or dissect. Just more complex.

Reply:
I was thinking more generally about the query support on runtime fields - I understand that prefix query isn't currently supported but was unsure why. Running a script over doc values to prefix query `dv.startsWith(qTerm)` doesn't seem like it would be slower than a term query `qTerm==dv`  (actually it would probably be faster? Less bytes to check).

Reply:
Oh! Your want to you why I only did this for the `term` query and not any
other others. Yeah, I think we would do it eventually. I just didn't do it
in the initial cut. I figured I'd start simple.